extendr-api/src/robj/from_robj.rs

```diff
@@ -289,3 +289,59 @@ where
         }
     }
 }
+
+#[cfg(feature = "faer")]
+impl<'a> FromRobj<'a> for faer::Mat<f64> {
+    fn from_robj(robj: &'a Robj) -> std::result::Result<Self, &'static str> {
+        use crate::scalar::Scalar;
+        if robj.is_matrix() {
```

Comment:
its worth noting that an R matrix is not always numeric. It can be character as well

Reply:
Won't the `as_real_slice`/`as_integer_slice` in that case return None?